cvpy/biomedimage/BiomedImage.py

```diff
@@ -241,3 +242,62 @@ def rename_columns():
 
         # Change column names in the ImageTable
         rename_columns()
+
+    def morphological_gradient(self, images: ImageTable, kernel_width: int = 3,
+                               kernel_height: int = 3, copy_vars: List[str] = None) -> ImageTable:
+        """
+        Compute morphological gradient for given 3D grayscale image table.
+
+        Parameters
+        ------------
+        images : :class:'ImageTable'
+            Specifies the image table that contains 3D grayscale image.
+        kernel_width : :class:'int'
+            Specifies the kernel width.
+        kernel_height : :class:'int'
+            Specifies the kernel height.
+        copy_vars : :class:'List[str]'
+            Specifies which columns to copy to the output image table.
+
+        Returns
+        ------------
+        :class:'ImageTable'
+        """
+
+        conn = self._cas_session
+        random_name_generator = RandomNameGenerator()
+
+        # Export images from 3d to 2d
+        name_image_2d = random_name_generator.generate_name()
+        image_2d = conn.CASTable(name=name_image_2d, replace=True)
+        conn.biomedimage.processbiomedimages(images=dict(table=images.table),
+                                             steps=[dict(stepparameters=dict(steptype='export'))],
+                                             casout=image_2d,
+                                             copyvars=copy_vars)
```

Comment:
It doesn't look like the copy_vars the user specifies here will be fully copied to the output table. You probably have to add these copy_vars to the next two actions as well. Since you are already copying '_biomedid_', '_biomeddimension_',  and '_sliceindex_' by default in processImages, you may need to add a little bit of logic to make sure you're not trying to copy the same columns twice. Something like this might work, where copyvars=process_images_copy_vars in your processImages action:

process_images_copy_vars = copy_vars
if '_biomedid_' not in process_images_copy_vars:
        process_images_copy_vars.append( '_biomedid_')
if '_biomeddimension_' not in process_images_copy_vars:
        process_images_copy_vars.append( '_biomeddimension_')
if '_sliceindex_' not in process_images_copy_vars:
        process_images_copy_vars.append( '_sliceindex_')

Reply:
Sure, let me do some experiments.